docs/examples/metadata_update/consumer_for_testing.py

```diff
@@ -0,0 +1,78 @@
+import asyncio
+import signal
+
+from rstream import (
+    AMQPMessage,
+    Consumer,
+    MessageContext,
+    OnClosedErrorInfo,
+    amqp_decoder,
+)
+
+STREAM = "my-test-stream"
+COUNT = 0
+consumer_closed = False
+
+
+async def print_count_values():
+
+    global COUNT
+
+    while True:
+        await asyncio.sleep(5)
+
+        print("NUMBER OF CONSUMED MESSAGES:         " + str(COUNT))
+
+
+async def consume():
+    asyncio.create_task(print_count_values())
+
+    async def connect() -> Consumer:
+        async def on_metadata_update(on_closed_info: OnClosedErrorInfo) -> None:
+            if on_closed_info.reason == "MetaData Update":
+                print(
+                    "metadata changed for stream : "
+                    + str(on_closed_info.streams[0])
+                    + " with code: "
+                    + on_closed_info.reason
+                )
+
+            else:
+                print(
+                    "connection has been closed from stream: "
+                    + str(on_closed_info.streams)
+                    + " for reason: "
+                    + str(on_closed_info.reason)
+                )
+
```

Comment:
Can you add a sleep before and add some retry? 
the event can happen during the restart, so restarting could take time.

Also testing the `streamNotExist` error code to close the consumer